TimeSeries endpoint fails even if Data-Entry-Date flag is not enabled

If a district deploys CDA (july 8th version) to their local T7 it will break the Time Series endpoint entirely even if they do not enable `data-entry-date` flag. 

Presumably the latest schema update will resolve this, but still thought it worth reporting! 

Error:

```
08-Jul-2025 16:08:48.742 WARNING [https-openssl-nio-155.88.11.74-8243-exec-5172] cwms.cda.ApiServlet.lambda$init$5 -3393660013003901610: Not Implemented for request: https://t7:8243/swt-data/timeseries?name=KEYS.Elev.Inst.1Hour.0.Ccp-Rev&office=SWT
        java.lang.UnsupportedOperationException: CWMS currently does not support the requested operation
                at org.jooq_3.18.7.ORACLE18C.debug(Unknown Source)
                at cwms.cda.data.dao.JooqDao.buildUnsupportedOperationException(JooqDao.java:653)
                at cwms.cda.data.dao.JooqDao.wrapException(JooqDao.java:268)
                at cwms.cda.data.dao.ExceptionWrappingListener.exception(ExceptionWrappingListener.java:12)
                at org.jooq.impl.ExecuteListeners.exception(ExecuteListeners.java:304)
                at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:361)
                at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:301)
                at org.jooq.impl.SelectImpl.fetchLazy(SelectImpl.java:2857)
                at org.jooq.impl.ResultQueryTrait.forEach(ResultQueryTrait.java:1458)
                at cwms.cda.data.dao.TimeSeriesDaoImpl.getTimeseries(TimeSeriesDaoImpl.java:429)
                at cwms.cda.api.TimeSeriesController.getAll(TimeSeriesController.java:449)
                at io.javalin.apibuilder.CrudFunction$1.invoke$lambda-0(CrudHandler.kt:30)
                at io.javalin.apibuilder.CrudFunctionHandler.handle(CrudHandler.kt)
                at cwms.cda.security.CdaAccessManager.manage(CdaAccessManager.java:37)
                at io.javalin.http.JavalinServlet.addHandler$lambda-0(JavalinServlet.kt:96)
                at io.javalin.http.JavalinServlet$lifecycle$2$1$1.invoke(JavalinServlet.kt:43)
                at io.javalin.http.JavalinServlet$lifecycle$2$1$1.invoke(JavalinServlet.kt:43)
                at io.javalin.http.JavalinServletHandler.executeNextTask(JavalinServletHandler.kt:99)
                at io.javalin.http.JavalinServletHandler.queueNextTaskOrFinish$lambda-1(JavalinServletHandler.kt:85)
                at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:995)
                at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2137)
                at io.javalin.http.JavalinServletHandler.queueNextTaskOrFinish$javalin(JavalinServletHandler.kt:85)
                at io.javalin.http.JavalinServletHandler.executeNextTask$lambda-11$lambda-10(JavalinServletHandler.kt:119)
                at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:616)
                at java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:628)
                at java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:1996)
                at io.javalin.http.JavalinServletHandler.executeNextTask(JavalinServletHandler.kt:119)
                at io.javalin.http.JavalinServletHandler.queueNextTaskOrFinish$lambda-1(JavalinServletHandler.kt:85)
                at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:995)
                at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2137)
                at io.javalin.http.JavalinServletHandler.queueNextTaskOrFinish$javalin(JavalinServletHandler.kt:85)
                at io.javalin.http.JavalinServletHandler.executeNextTask$lambda-11$lambda-10(JavalinServletHandler.kt:119)
                at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:616)
                at java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:628)
                at java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:1996)
                at io.javalin.http.JavalinServletHandler.executeNextTask(JavalinServletHandler.kt:119)
                at io.javalin.http.JavalinServletHandler.queueNextTaskOrFinish$lambda-1(JavalinServletHandler.kt:85)
                at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:995)
                at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2137)
                at io.javalin.http.JavalinServletHandler.queueNextTaskOrFinish$javalin(JavalinServletHandler.kt:85)
                at io.javalin.http.JavalinServlet.service(JavalinServlet.kt:89)
                at javax.servlet.http.HttpServlet.service(HttpServlet.java:623)
                at cwms.cda.ApiServlet.service(ApiServlet.java:1010)
                at javax.servlet.http.HttpServlet.service(HttpServlet.java:623)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:199)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:144)
                at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:168)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:144)
                at org.apache.catalina.filters.HttpHeaderSecurityFilter.doFilter(HttpHeaderSecurityFilter.java:129)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:168)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:144)
                at org.apache.catalina.filters.CorsFilter.handleNonCORS(CorsFilter.java:333)
                at org.apache.catalina.filters.CorsFilter.doFilter(CorsFilter.java:160)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:168)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:144)
                at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168)
                at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
                at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:597)
                at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130)
                at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:660)
                at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
                at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
                at org.apache.catalina.authenticator.SingleSignOn.invoke(SingleSignOn.java:241)
                at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
                at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:346)
                at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:396)
                at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
                at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:937)
                at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1793)
                at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
                at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
                at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
                at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
                at java.lang.Thread.run(Thread.java:750)
        Caused by: java.sql.SQLSyntaxErrorException: ORA-00904: "CWMS_20"."CWMS_TS"."RETRIEVE_TS_ENTRY_OUT_TAB": invalid identifier

                at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:509)
                at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:461)
                at oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:1104)
                at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:550)
                at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:268)
                at oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:655)
                at oracle.jdbc.driver.T4CPreparedStatement.doOall8(T4CPreparedStatement.java:270)
                at oracle.jdbc.driver.T4CPreparedStatement.doOall8(T4CPreparedStatement.java:91)
                at oracle.jdbc.driver.T4CPreparedStatement.executeForDescribe(T4CPreparedStatement.java:807)
                at oracle.jdbc.driver.OracleStatement.executeMaybeDescribe(OracleStatement.java:983)
                at oracle.jdbc.driver.OracleStatement.doExecuteWithTimeout(OracleStatement.java:1168)
                at oracle.jdbc.driver.OraclePreparedStatement.executeInternal(OraclePreparedStatement.java:3666)
                at oracle.jdbc.driver.T4CPreparedStatement.executeInternal(T4CPreparedStatement.java:1426)
                at oracle.jdbc.driver.OraclePreparedStatement.execute(OraclePreparedStatement.java:3778)
                at oracle.jdbc.driver.OraclePreparedStatementWrapper.execute(OraclePreparedStatementWrapper.java:1081)
                at sun.reflect.GeneratedMethodAccessor186.invoke(Unknown Source)
                at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                at java.lang.reflect.Method.invoke(Method.java:498)
                at org.apache.tomcat.jdbc.pool.StatementFacade$StatementProxy.invoke(StatementFacade.java:123)
                at com.sun.proxy.$Proxy129.execute(Unknown Source)
                at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:219)
                at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:4695)
                at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:236)
                at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:346)
                ... 69 more
        Caused by: Error : 904, Position : 135, Sql = select DATE_TIME "DATE_TIME", VALUE "VALUE", "CWMS_20"."CWMS_TS"."NORMALIZE_QUALITY"(nvl("QUALITY_CODE", 5)) "QUALITY_NORM" from table(cwms_20.cwms_ts.retrieve_ts_entry_out_tab("CWMS_20"."CWMS_TS"."GET_TS_ID"(:1 , "CWMS_20"."CWMS_UTIL"."GET_DB_OFFICE_ID"(:2 )),"CWMS_20"."CWMS_UTIL"."GET_DEFAULT_UNITS"("CWMS_20"."CWMS_TS"."GET_BASE_PARAMETER_ID"("CWMS_20"."CWMS_TS"."GET_TS_CODE"(:3 , "CWMS_20"."CWMS_UTIL"."GET_DB_OFFICE_ID"(:4 ))), :5 ),cwms_20.cwms_util.to_timestamp(:6 ), cwms_20.cwms_util.to_timestamp(:7 ), 'UTC',:8 ,:9 ,:10 ,:11 ,:12 ,:13 ,:14 ,"CWMS_20"."CWMS_UTIL"."GET_DB_OFFICE_ID"(:15 )) ) retrieveTs where ("DATE_TIME" >= "CWMS_20"."CWMS_UTIL"."TO_TIMESTAMP"(nvl(:16 , :17 )) and "DATE_TIME" <= "CWMS_20"."CWMS_UTIL"."TO_TIMESTAMP"(:18 )) fetch next :19  rows only, OriginalSql = select DATE_TIME "DATE_TIME", VALUE "VALUE", "CWMS_20"."CWMS_TS"."NORMALIZE_QUALITY"(nvl("QUALITY_CODE", 5)) "QUALITY_NORM" from table(cwms_20.cwms_ts.retrieve_ts_entry_out_tab("CWMS_20"."CWMS_TS"."GET_TS_ID"(?, "CWMS_20"."CWMS_UTIL"."GET_DB_OFFICE_ID"(?)),"CWMS_20"."CWMS_UTIL"."GET_DEFAULT_UNITS"("CWMS_20"."CWMS_TS"."GET_BASE_PARAMETER_ID"("CWMS_20"."CWMS_TS"."GET_TS_CODE"(?, "CWMS_20"."CWMS_UTIL"."GET_DB_OFFICE_ID"(?))), ?),cwms_20.cwms_util.to_timestamp(?), cwms_20.cwms_util.to_timestamp(?), 'UTC',?,?,?,?,?,?,?,"CWMS_20"."CWMS_UTIL"."GET_DB_OFFICE_ID"(?)) ) retrieveTs where ("DATE_TIME" >= "CWMS_20"."CWMS_UTIL"."TO_TIMESTAMP"(nvl(?, ?)) and "DATE_TIME" <= "CWMS_20"."CWMS_UTIL"."TO_TIMESTAMP"(?)) fetch next ? rows only, Error Msg = ORA-00904: "CWMS_20"."CWMS_TS"."RETRIEVE_TS_ENTRY_OUT_TAB": invalid identifier

                at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:513)
                ... 92 more
```

--- a/cwms-data-api/src/main/java/cwms/cda/data/dao/TimeSeriesDaoImpl.java
+++ b/cwms-data-api/src/main/java/cwms/cda/data/dao/TimeSeriesDaoImpl.java
@@ -456,7 +456,7 @@ public class TimeSeriesDaoImpl extends JooqDao<TimeSeries> implements TimeSeries
         );
 
         String retrievalMethod;
-        if (entryDateSupport != null) {
+        if (includeEntryDate) {
             retrievalMethod = "cwms_20.cwms_ts.retrieve_ts_entry_out_tab";  // New method that supports entry date
         } else {
             retrievalMethod = "cwms_20.cwms_ts.retrieve_ts_out_tab";    // Legacy method without entry date
@@ -508,7 +508,7 @@ public class TimeSeriesDaoImpl extends JooqDao<TimeSeries> implements TimeSeries
 
             if (pageSize > 0) {
                 query.limit(DSL.val(pageSize + 1));
-                if (entryDateSupport != null) {
+                if (includeEntryDate) {
                     query2.limit(DSL.val(pageSize + 1));
                 }
             }
@@ -516,7 +516,7 @@ public class TimeSeriesDaoImpl extends JooqDao<TimeSeries> implements TimeSeries
             if (requestParameters.isIncludeEntryDate()) {
                 logger.fine(() -> query2.getSQL(ParamType.INLINED));
                 try (Cursor<Record4<Timestamp, Double, BigDecimal, Timestamp>> recCursor = query2.fetchLazy()) {
-                    for(Record tsRecord: recCursor){
+                    for (Record tsRecord: recCursor) {
                         timeseries.addValue(
                                 tsRecord.getValue(dateTimeCol),
                                 tsRecord.getValue(valueCol),
@@ -527,7 +527,7 @@ public class TimeSeriesDaoImpl extends JooqDao<TimeSeries> implements TimeSeries
             } else {
                 logger.fine(() -> query.getSQL(ParamType.INLINED));
                 try (Cursor<Record3<Timestamp, Double, BigDecimal>> recCursor = query.fetchLazy()) {
-                    for(Record tsRecord: recCursor){
+                    for (Record tsRecord: recCursor) {
                         timeseries.addValue(
                                 tsRecord.getValue(dateTimeCol),
                                 tsRecord.getValue(valueCol),

--- a/cwms-data-api/src/main/java/cwms/cda/data/dao/TimeSeriesDaoImpl.java
+++ b/cwms-data-api/src/main/java/cwms/cda/data/dao/TimeSeriesDaoImpl.java
@@ -172,14 +172,12 @@ public class TimeSeriesDaoImpl extends JooqDao<TimeSeries> implements TimeSeries
         Timestamp tsCursor = null;
         Integer total = null;
 
-        if (includeEntryDate) {
-            Record entryDateSupport = dsl.select(asterisk()).from(table("ALL_TYPES"))
-                    .where(field("TYPE_NAME").eq("ZTSV_ENTRY_TYPE"))
-                    .and(field("OWNER").eq("CWMS_20")).fetchOne();
+        Record entryDateSupport = dsl.select(asterisk()).from(table("ALL_TYPES"))
+            .where(field("TYPE_NAME").eq("ZTSV_ENTRY_TYPE"))
+            .and(field("OWNER").eq("CWMS_20")).fetchOne();
 
-            if (entryDateSupport == null) {
-                throw new DataAccessException("Data entry date retrieval is not supported by this database");
-            }
+        if (includeEntryDate && entryDateSupport == null) {
+            throw new DataAccessException("Data entry date retrieval is not supported by this database");
         }
 
         if (page != null && !page.isEmpty()) {
@@ -364,11 +362,18 @@ public class TimeSeriesDaoImpl extends JooqDao<TimeSeries> implements TimeSeries
             }
         );
 
+        String retrievalMethod;
+        if (entryDateSupport != null) {
+            retrievalMethod = "cwms_20.cwms_ts.retrieve_ts_entry_out_tab";  // New method that supports entry date
+        } else {
+            retrievalMethod = "cwms_20.cwms_ts.retrieve_ts_out_tab";    // Legacy method without entry date
+        }
+
         // Now we're going to call the retrieve_ts_entry_out_tab function to get the data and build an
         // internal table from it so we can manipulate it further
         // This code assumes the database timezone is in UTC (per Oracle recommendation)
         SQL retrieveSelectData = DSL.sql(
-                "table(cwms_20.cwms_ts.retrieve_ts_entry_out_tab(?,?,"
+                "table(" + retrievalMethod + "(?,?,"
                         + "cwms_20.cwms_util.to_timestamp(?), cwms_20.cwms_util.to_timestamp(?), 'UTC',"
                         + "?,?,?,?,?,"
                         + getVersionPart(versionDate) + ",?,?) ) retrieveTs",
@@ -410,7 +415,9 @@ public class TimeSeriesDaoImpl extends JooqDao<TimeSeries> implements TimeSeries
 
             if (pageSize > 0) {
                 query.limit(DSL.val(pageSize + 1));
-                query2.limit(DSL.val(pageSize + 1));
+                if (entryDateSupport != null) {
+                    query2.limit(DSL.val(pageSize + 1));
+                }
             }
 
             if (includeEntryDate) {

--- a/cwms-data-api/src/test/java/cwms/cda/api/TimeseriesControllerTestIT.java
+++ b/cwms-data-api/src/test/java/cwms/cda/api/TimeseriesControllerTestIT.java
@@ -17,6 +17,7 @@ import com.fasterxml.jackson.databind.ObjectMapper;
 import cwms.cda.ApiServlet;
 import cwms.cda.formatters.Formats;
 import fixtures.CwmsDataApiSetupCallback;
+import fixtures.MinimumSchema;
 import fixtures.TestAccounts;
 import io.restassured.RestAssured;
 import io.restassured.filter.log.LogDetail;
@@ -41,6 +42,8 @@ import org.junit.jupiter.params.provider.EnumSource;
 
 @Tag("integration")
 class TimeseriesControllerTestIT extends DataApiTestIT {
+    public static final int MINIMUM_SCHEMA = 999999;
+
     @Test
     void test_lrl_timeseries_psuedo_reg1hour() throws Exception {
         ObjectMapper mapper = new ObjectMapper();
@@ -108,6 +111,7 @@ class TimeseriesControllerTestIT extends DataApiTestIT {
     }
 
     @Test
+    @MinimumSchema(MINIMUM_SCHEMA)
     void test_local_regular_new_LRTS_ID() throws Exception {
         ObjectMapper mapper = new ObjectMapper();
 
@@ -380,6 +384,7 @@ class TimeseriesControllerTestIT extends DataApiTestIT {
     }
 
     @Test
+    @MinimumSchema(MINIMUM_SCHEMA)
     void test_lrl_1day_max_version_with_entry_date() throws Exception {
         ObjectMapper mapper = new ObjectMapper();
 
@@ -709,6 +714,7 @@ class TimeseriesControllerTestIT extends DataApiTestIT {
     }
 
     @Test
+    @MinimumSchema(MINIMUM_SCHEMA)
     void test_include_data_entry_date() throws Exception {
         ObjectMapper mapper = new ObjectMapper();
 

--- a/cwms-data-api/src/test/java/cwms/cda/api/TimeseriesControllerTestIT.java
+++ b/cwms-data-api/src/test/java/cwms/cda/api/TimeseriesControllerTestIT.java
@@ -909,6 +909,7 @@ class TimeseriesControllerTestIT extends DataApiTestIT {
     }
 
     @Test
+    @MinimumSchema(MINIMUM_SCHEMA)
     void test_attempt_store_with_entry_date() throws Exception
     {
         ObjectMapper mapper = new ObjectMapper();
@@ -1295,6 +1296,7 @@ class TimeseriesControllerTestIT extends DataApiTestIT {
     }
 
     @Test
+    @MinimumSchema(MINIMUM_SCHEMA)
     void test_lrl_trim_with_data_entry_date() throws Exception {
         ObjectMapper mapper = new ObjectMapper();
 

